pkg/app/accounts/basic.go

```diff
@@ -44,6 +99,12 @@ func (u *HTTPBasicUser) Username() string { return u.username }
 func (u *HTTPBasicUser) Email() string { return "" }
 
 func userFromRequest(r *http.Request) (*HTTPBasicUser, error) {
+	// Also accepts putting the username in a cookie to support using a
+	// browser to interact with CO.
+	cookie, err := r.Cookie(unameCookie)
+	if err == nil && cookie.Value != "" {
+		return &HTTPBasicUser{cookie.Value}, nil
+	}
 	username, _, ok := r.BasicAuth()
 	if !ok {
 		return nil, apperr.NewBadRequestError("No username in request", nil)
```

Comment:
as we discussed, it could be redirect to login page.

Reply:
No problem, but I would prefer to do this by another PR since this would need to change the interface of `UserFromRequest` by passing `http.ResponseWriter` to it.